app/controllers/api/v1/search_controller.rb

```diff
@@ -294,11 +294,24 @@ def index
         # perform TDR search, if enabled, and there are facets/terms provided by user
         if User.feature_flag_for_instance(current_api_user, 'cross_dataset_search_backend') && (@facets.present? || @term_list.present?)
           @tdr_results = self.class.get_tdr_results(selected_facets: @facets, terms: @term_list)
+          
+          if (@facets.present?)
+            simple_tdr_results = self.class.simplify_TDR_facet_search_results(@tdr_results, @facets)
+            matched_tdr_studies = self.class.match_studies_by_facet(simple_tdr_results, @facets)
+            
+            if @studies_by_facet.present?
+              @studies_by_facet = @studies_by_facet.merge(matched_tdr_studies)
+            else
+              @studies_by_facet = matched_tdr_studies
+            end
+          end
           # just log for now
           logger.info "Found #{@tdr_results.keys.size} results in Terra Data Repo"
           logger.info pp @tdr_results if @tdr_results.any?
           @tdr_results.each do |short_name, tdr_result|
-            @studies << tdr_result
+              if tdr_result[:accession] != nil
```

Comment:
How could `:accession` be `nil` for a TDR result?  Shouldn't we always get back an HCA short name for all TDR results from `process_tdr_result_row`?